src/core/component.js

```diff
@@ -212,7 +215,7 @@ module.exports.registerComponent = function (name, definition) {
   });
 
   if (components[name]) {
-    throw new Error('The component ' + name + ' has been already registered');
+    warn('The component ' + name + ' has been already registered');
```

Comment:
OK. It's usually a benign operation and a warning usually suffice. Halting everything is pretty harsh. But I'll just leave it as an error until it becomes a problem.